### PR TITLE
[1.20.2] Rename "Main mod protocol" -> "modpack protocol", add documentation

### PR DESCRIPTION
**Pull request by TheGlitch76**
This is technically a breaking change, but in an API marked `@ApiStatus.Experimental` and, as far as I'm aware, entirely unused because it has been, up to this point, undocumented.

Intended to be merged when 1.20.1 development begins

### CODE REVIEW

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java
Comment (EnnuiL) [inline library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java]:
```suggestion
 * is otherwise identical to standard mod protocols.
```

**Review by EnnuiL — CHANGES_REQUESTED** [review]
pass 1 of my review:

**Review by OroArmor — APPROVED** [review]

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java
Comment (EnnuiL) [inline library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java]:
```suggestion
 * <pre>{@code
```

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java
Comment (EnnuiL) [inline library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java]:
```suggestion
 * <pre>{@code
```

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java
Comment (EnnuiL) [inline library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java:115]:
```suggestion
	 * @param player the player to check against
	 * @param modContainer the mod container defining the protocol version
	 * @return -1 if not supported, or the latest mod protocol version supported by the player otherwise.
```

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java
Comment (EnnuiL) [inline library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java]:
```suggestion
	 * {@return a string representing the modpack protocol id, or {@code null} otherwise}
```

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java
Comment (EnnuiL) [inline library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java]:
```suggestion
	 * {@return {@code true} if a modpack protocol has been provided through the config, or {@code false} otherwise}
```

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java
Comment (EnnuiL) [inline library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java]:
```suggestion
	 * {@return a string representing the modpack protocol display name, or {@code null} otherwise}
```

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java
Comment (EnnuiL) [inline library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java:127]:
```suggestion
	 * @param handler the player's network handler to check against
	 * @param modContainer the mod container defining the protocol version
	 * @return -1 if not supported, or the latest mod protocol version supported by the player otherwise.
```

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/impl/RegistryConfig.java
Comment (EnnuiL) [inline library/core/registry/src/main/java/org/quiltmc/qsl/registry/impl/RegistryConfig.java]:
```suggestion
		@Comment("The modpack protocol is a feature allowing you to prevent clients with mismatched protocol versions from joining.")
```

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/impl/sync/mod_protocol/ModProtocolDef.java
Comment (EnnuiL) [inline library/core/registry/src/main/java/org/quiltmc/qsl/registry/impl/sync/mod_protocol/ModProtocolDef.java]:
```suggestion
 * A definition of a mod protocol.
 *
 * @param id the id of the protocol. For mods, this always matches the mod id.
```

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/impl/sync/mod_protocol/ModProtocolDef.java
Comment (EnnuiL) [inline library/core/registry/src/main/java/org/quiltmc/qsl/registry/impl/sync/mod_protocol/ModProtocolDef.java]: maybe?
```suggestion
 * @param optional whether a player can connect to a server with any protocol version. The player's maximum supported protocol version can still be queried with {@link ModProtocols#getSupported(ServerPlayerEntity, ModContainer)}
```

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/impl/sync/mod_protocol/ModProtocolDef.java
Comment (EnnuiL) [inline library/core/registry/src/main/java/org/quiltmc/qsl/registry/impl/sync/mod_protocol/ModProtocolDef.java:34]: This is missing a description

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java
Comment (EnnuiL) [inline library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java:149]:
```suggestion
	 * Checks the modpack protocol version supported by a given player. Returns -1 if the server and player's
	 * {@linkplain #getModpackProtocolId() modpack protocol id} do not match.
	 *
	 * @param handler the player's network handler to check against
	 * @return -1 if not supported, or the latest modpack protocol version supported by the player otherwise.
```
maybe...? i'd double-check this

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java
Comment (EnnuiL) [inline library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java:138]:
```suggestion
	 * Checks the modpack protocol version supported by a given player. Returns -1 if the server and player's
	 * {@linkplain #getModpackProtocolId() modpack protocol id} do not match.
	 *
	 * @param player the player to check against
	 * @return -1 if not supported, or the latest modpack protocol version supported by the player otherwise.
```
hm, there is a bad copy-and-paste going on here

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java
Comment (EnnuiL) [inline library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java]:
```suggestion
	public static int getSupportedModpack(@NotNull ServerPlayerEntity player) {
		return getSupportedModpack(player.networkHandler);
	}
```

**Review by EnnuiL — CHANGES_REQUESTED** [review]
I'd recommending waiting until PR #341 is merged, since that simplifies a lot of the config code

File: library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java
Comment (TheGlitch76) [reply library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java:115]: Since we're being picky, I'm going to do this one by hand when I fix the build since i disagree that the special-case should come first on the `@return`

**Review by OroArmor — APPROVED** [review]

**Comment by eristhea** [comment]
> Intended to be merged when 1.20.1 development begins

literally heartbreaking

**Pull request closed without merging**
